plugins/ua_pubsub_realtime_etf.c

```diff
@@ -0,0 +1,259 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+ * Copyright (c) 2019-2020 Kalycito Infotech Private Limited
+ */
+
+/* This file contains socket address handling and tx time calculation for real time publish */
+
+#include <open62541/plugin/network.h>
+#include <open62541/plugin/log_stdout.h>
+#include <include/open62541/plugin/pubsub_realtime_etf.h>
+#ifdef UA_ENABLE_PUBSUB_ETH_UADP
+#include <linux/if_packet.h>
+#endif
+#include "poll.h"
+#include "linux/errqueue.h"
+#include "time.h"
+
+#define   MICRO_SECONDS                                   1000
+#define   MILLI_SECONDS                                   1000 * 1000
+#define   SECONDS                                         1000 * 1000 * 1000
+#define   CYCLE_TIME                                      250 * 1000
+#define   QBV_OFFSET                                      25
+#define   SECONDS_INCREMENT                               1
+#define   FAILURE_EXIT                                    -1
+#define   SHIFT_32BITS                                    32
+
+struct timespec        nextCycleStartTime;
```

Comment:
Global variables in a plugin are a no-go.
You (Kalycito) have requests coming in for multiple publishers with different cycle times in one application.
This cannot be done with the cycle time in a global variable.